Kan 17 fully swap static data into data from db

Ech0ke opened the pull request:
Data is now from db. **Currently there is no refresh when creating/editing note, so make sure you manually add folder for your userID according to this example in firestore**. Make sure to create new user, since we are now on v3 and not v2 anymore.
![image](https://github.com/ISI-dotnet/notes-app/assets/99315244/87469914-0b02-4580-9387-205067c9249c)


By going into folder and navigating back, you will trigger refetch and you should see new/updated notes in **browse** tab.

You can test the functionality

- Create note
- Edit note
- Nested folders

I will now start implementing ability to create folders from the app and will probably add live data refresh or at least pull up gesture to refresh (refetch).

matuliux commented on src/app/(tabs)/index.tsx:
```diff
@@ -2,14 +2,33 @@ import { auth } from "@/firebaseConfig"
   return (
     <SafeAreaView className="flex-1">
-      <Text>Welcome {user?.getIdToken}</Text>
+      <Text>Welcome {username}</Text>
```
Here I get `Welcome {my jwt token}`
I dont think this is a big issue but just wanna check if you intended that

matuliux reviewed (commented):
I have added a new folder into db
![image](https://github.com/ISI-dotnet/notes-app/assets/100164021/2a9a7e81-49b6-40af-a8da-b61f87ff97df)

and assigned a note to it

![image](https://github.com/ISI-dotnet/notes-app/assets/100164021/c3f503af-b3db-4ad8-9547-15f94412c271)

But after reloading the app I dont see neither the note nor the folder

![image](https://github.com/ISI-dotnet/notes-app/assets/100164021/3d842f00-569d-4a89-b939-bf36bf790941)

Can you spot why?

matuliux reviewed (commented):
Also I think this part should not be here. I have had it removed with my previous PR just not sure if changes got lost or if they were reverted

![image](https://github.com/ISI-dotnet/notes-app/assets/100164021/9ed78f4c-572a-41ce-a600-84c0099cffbe)

Edit: I added this part into my upcoming PR so you can leave it as is


![image](https://github.com/ISI-dotnet/notes-app/assets/100164021/929fa2dd-a8d0-47f1-bffa-8df0f5c32597)

Ech0ke commented:
@matuliux 
I've fixed folder fetching, refactored auth according to expo documentation and now upon reloading app, you don't need to relogin, since auth state is saved.

try to run it now but there may be cache left, so after pulling the changes, run this command: `npx expo start --tunnel --clear`

matuliux reviewed (approved):
Got folders working (issue from before seems to be fixed)
Everything works great!
Also I haven't noticed any flickering yet, not even when typing big notes. If you get this bug please send a screen recording so I could try to recreate it
